internal/services/qumulo/qumulo_file_system_resource.go

```diff
@@ -0,0 +1,304 @@
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Qumulo.FileSystemsClient
+
+			id, err := filesystems.ParseFileSystemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(*id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			var config FileSystemResourceSchema
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			config.Name = id.FileSystemName
+			config.ResourceGroupName = id.ResourceGroupName
+
+			if model := resp.Model; model != nil {
+				// model.Properties and model.Properties.MarketplaceDetails is not pointer, so we don't need to check nil
+				config.Zone = pointer.From(model.Properties.AvailabilityZone)
+				config.PlanId = model.Properties.MarketplaceDetails.PlanId
+				config.StorageSku = model.Properties.StorageSku
+				config.Location = location.Normalize(model.Location)
+				config.Tags = pointer.From(model.Tags)
+
+				subnetId, err := commonids.ParseSubnetIDInsensitively(model.Properties.DelegatedSubnetId)
+				if err != nil {
+					return err
+				}
+				config.SubnetId = subnetId.ID()
+			}
+
+			return metadata.Encode(&config)
+		},
+	}
+}
+
+func (r FileSystemResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Qumulo.FileSystemsClient
+
+			id, err := filesystems.ParseFileSystemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r FileSystemResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 60 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Qumulo.FileSystemsClient
+
+			id, err := filesystems.ParseFileSystemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config FileSystemResourceSchema
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			payload := filesystems.LiftrBaseStorageFileSystemResourceUpdate{}
+
+			if metadata.ResourceData.HasChange("tags") {
+				payload.Tags = pointer.To(config.Tags)
+			}
+
+			if _, err := client.Update(ctx, *id, payload); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func checkSubnet(ctx context.Context, rawSubnetId string, metadata sdk.ResourceMetaData) error {
+	const (
+		delegationAction = "Microsoft.Network/virtualNetworks/subnets/join/action"
+		delegationName   = "Qumulo.Storage/fileSystems"
+	)
```

Comment:
Do these need to be defined as constants?

Reply:
changed to string variable.